File: API/demo/src/main/java/com/example/demo/model/UserSetting.java
package com.example.demo.model;

import jakarta.persistence.*;

@Entity
public class UserSetting {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    @JoinColumn(name = "usuario_id")
    private Users usuario;
    private boolean notificaciones = true;
    private String sonido = "default";

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Users getUsuario() {
        return usuario;
    }

    public void setUsuario(Users usuario) {
        this.usuario = usuario;
    }

    public boolean isNotificaciones() {
        return notificaciones;
    }

    public void setNotificaciones(boolean notificaciones) {
        this.notificaciones = notificaciones;
    }

    public String getSonido() {
        return sonido;
    }

    public void setSonido(String sonido) {
        this.sonido = sonido;
    }
}

File: Aplicacion Movil/MedMonitor/app/src/main/java/Models/UserSetting.kt
package Models

import com.google.gson.annotations.SerializedName

data class UserSetting(
    @SerializedName("id") val id: Long,
    @SerializedName("usuario_id") val usuarioId: Int,
    @SerializedName("notificaciones") val notificaciones: Boolean,
    @SerializedName("sonido") val sonido: String
)
